### DISCUSSION TITLE
How to specify which is the admin node?

### DISCUSSION THREAD

**Discussion by tutiplain** [aergoio/aergolite#8] [Q&A] [answered]
I have managed to successfully compile the library, and to generate a public and private key for the node I want to use as the blockchain administrator. I have moved from using the sqlite shell to using the library in go itself.  I am opening my database with the following:

`file:test.db?blockchain=on&&discovery=local:7243&admin=public_key`

I am then running a PRAGMA blockchain_status to see if my node meant to be my blockchain administrator is authorized. I see the following:

`{
  "use_blockchain": true,

  "blockchain": {
    "last_block": 0,
    "state_hash": "E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855",
    "integrity": {
      "state": "empty",
      "chain": "pruned"
    }
  },

  "node": {
    "id": 127557326,
    "pubkey": "AmGcjenNZ9U5K9shW13wYVF9RgG46ctykA7bvysFQajyarCeBAYo",
    "type": "unauthorized",
    "local_transactions": {
      "processed": 0,
      "unprocessed": 0,
      "total": 0
    }
  },

  "mempool": {
    "num_transactions": 0
  },

  "network": {
    "num_authorized_nodes": 0,
    "num_connected_peers": 0
  },

  "downstream_state": "unknown",
  "upstream_state": "unknown"
}`

The node appears as unauthorized. I know that I'm supposed to use PRAGMA add_node to add new_nodes, but how do I get the first one to be authorized?

**Comment by kroggen** [comment, marked as answer]
There is no "admin node". There is an "admin user".

This means that the blockchain administrator can manage the network from any of the participating nodes. At least after the network is started.

At bootstrapping (initial setup):

The first node to be authorized must be the one in which the command is being executed.

In your case you get the pubkey from the blockchain_status, and then construct an `add_node` command:

```
pragma add_node="AmGcjenNZ9U5K9shW13wYVF9RgG46ctykA7bvysFQajyarCeBAYo"
```

Then you can authorize other nodes on the network.

For this you run `pragma nodes` to list them, and then `pragma add_node`

This must be done in a node that are already authorized.

The above command will call the transaction signature callback function (`on_sign_transaction`) where the transaction must be signed using the blockchain administrator private key.
